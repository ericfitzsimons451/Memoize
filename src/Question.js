import React from 'react';
import './reset.css'
import './Question.scss';

function Question(props){
  return (
    <div className="question-card">
      <h2 className="question-card--topic">{props.questionInfo.question}</h2>
      <h3 className="answer-selection-one">{props.questionInfo.answer1}</h3>
      <h3 className="answer-selection-two">{props.questionInfo.answer2}</h3>
      <h3 className="answer-selection-three">{props.questionInfo.answer3}</h3>
      <button onClick={props.showAnswer} type="button" >button</button>
    </div>
  )
}

export default Question;