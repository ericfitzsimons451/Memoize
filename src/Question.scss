.question-card {
  width: 400px;
  border: 1px solid black;
  background-color: #778078;
  background-image: url("https://www.transparenttextures.com/patterns/asfalt-dark.png");
  display: flex;
  flex-direction: column;
  margin: 5px;
  border-radius: 20px;
}

.question-card--topic {
  height: 25%;
  font-size: 1.7rem;
  padding: 7px 0px 20px 0px;
}

.answer-selection-one,
.answer-selection-two,
.answer-selection-three {
  font-size: 1.5rem;
  padding: 10px 15px 10px 15px;
  border-bottom: 1px dashed white;
  border-radius: 20px;
  background-color: #6B3137;
  margin: 5px;
  height: 18.5%;
  &:hover {
    cursor: pointer;
    background-color: #F19143;
  }
}

.submit-answer-button {
  height: 18.5%;
  font-size: 1.5rem;
  background-color: white;
  border-radius: 20px;
  margin: 5px;
  &:hover {
    cursor: pointer;
  }
}
