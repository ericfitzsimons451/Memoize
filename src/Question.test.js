import React from 'react';
import Question from './Question.js';
import { shallow } from 'enzyme';

const mockQuestion = { "id": 1,
    "categories":"Context",
    "question": "What is the default value of the THIS keyword?",
    "answer1": "The DOM Tree",
    "answer2": "The Global Window Object", 
    "answer3": "The Call Stack",
    "correctAnswer": "The Global Window Object",
    "link": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this"
  }

const mockVerifyAnswer = jest.fn()

describe('Question', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Question questionInfo={mockQuestion}
                verifyAnswer={mockVerifyAnswer}
                key={mockQuestion.id} />
      )
  })

  it('should match the snapshot with all the data passed down', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should register a click when any H3 element is clicked', () => {
    wrapper.find('.answer-select-one', 'answer-select-two', 'answer-select-three', () => {
      expect(mockVerifyAnswer).toBeCalled()
    })

 
  })
})