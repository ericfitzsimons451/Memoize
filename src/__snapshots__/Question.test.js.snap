// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should match the snapshot with all the data passed down 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Question
    questionInfo={
      Object {
        "answer1": "The DOM Tree",
        "answer2": "The Global Window Object",
        "answer3": "The Call Stack",
        "categories": "Context",
        "correctAnswer": "The Global Window Object",
        "id": 1,
        "link": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this",
        "question": "What is the default value of the THIS keyword?",
      }
    }
    verifyAnswer={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="question-card--topic"
        >
          1
          ) 
          What is the default value of the THIS keyword?
        </h2>,
        <h3
          className="answer-selection-one"
          onClick={[MockFunction]}
        >
          The DOM Tree
        </h3>,
        <h3
          className="answer-selection-two"
          onClick={[MockFunction]}
        >
          The Global Window Object
        </h3>,
        <h3
          className="answer-selection-three"
          onClick={[MockFunction]}
        >
          The Call Stack
        </h3>,
      ],
      "className": "question-card",
      "id": 1,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            1,
            ") ",
            "What is the default value of the THIS keyword?",
          ],
          "className": "question-card--topic",
        },
        "ref": null,
        "rendered": Array [
          1,
          ") ",
          "What is the default value of the THIS keyword?",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The DOM Tree",
          "className": "answer-selection-one",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "The DOM Tree",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The Global Window Object",
          "className": "answer-selection-two",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "The Global Window Object",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The Call Stack",
          "className": "answer-selection-three",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "The Call Stack",
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="question-card--topic"
          >
            1
            ) 
            What is the default value of the THIS keyword?
          </h2>,
          <h3
            className="answer-selection-one"
            onClick={[MockFunction]}
          >
            The DOM Tree
          </h3>,
          <h3
            className="answer-selection-two"
            onClick={[MockFunction]}
          >
            The Global Window Object
          </h3>,
          <h3
            className="answer-selection-three"
            onClick={[MockFunction]}
          >
            The Call Stack
          </h3>,
        ],
        "className": "question-card",
        "id": 1,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              1,
              ") ",
              "What is the default value of the THIS keyword?",
            ],
            "className": "question-card--topic",
          },
          "ref": null,
          "rendered": Array [
            1,
            ") ",
            "What is the default value of the THIS keyword?",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The DOM Tree",
            "className": "answer-selection-one",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "The DOM Tree",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The Global Window Object",
            "className": "answer-selection-two",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "The Global Window Object",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The Call Stack",
            "className": "answer-selection-three",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "The Call Stack",
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
